File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03408/p03408_num7_parsed.py
# Reading inputs
N = int(input())
blue_cards = [input() for _ in range(N)]
M = int(input())
red_cards = [input() for _ in range(M)]

# Counting the occurrences of each string in blue and red cards
blue_count = {x: blue_cards.count(x) for x in set(blue_cards)}
red_count = {x: red_cards.count(x) for x in set(red_cards)}

# Calculating the maximum balance
max_balance = 0
for string, count in blue_count.items():
    balance = count - red_count.get(string, 0)
    max_balance = max(max_balance, balance)

# Printing the maximum balance
print(max_balance)

File: Datasets/TrickyBugs/GenProgs/tc_generated_progs_cpp/p03408/p03408_num7_parsed.cpp
#include <iostream>
#include <map>
using namespace std;

int main() {
    int N;
    cin >> N;
    
    map<string, int> m;
    for (int i = 0; i < N; ++i) {
        string s;
        cin >> s;
        ++m[s];
    }
    
    int M;
    cin >> M;
    
    for (int i = 0; i < M; ++i) {
        string t;
        cin >> t;
        --m[t];
    }
    
    int max_balance = 0;
    for (const auto& p : m) {
        max_balance = max(max_balance, p.second);
    }
    
    cout << max_balance << endl;
    return 0;
}
